diff --git a/node_modules/@react-native-community/progress-view/common/cpp/react/renderer/components/progressview/RNCProgressViewComponentDescriptors.h b/node_modules/@react-native-community/progress-view/common/cpp/react/renderer/components/progressview/RNCProgressViewComponentDescriptors.h
index 0aeef83..cfd4250 100644
--- a/node_modules/@react-native-community/progress-view/common/cpp/react/renderer/components/progressview/RNCProgressViewComponentDescriptors.h
+++ b/node_modules/@react-native-community/progress-view/common/cpp/react/renderer/components/progressview/RNCProgressViewComponentDescriptors.h
@@ -18,15 +18,15 @@ class RNCProgressViewComponentDescriptor final
       : ConcreteComponentDescriptor(parameters),
         imageManager_(std::make_shared<ImageManager>(contextContainer_)) {}
 
-  void adopt(ShadowNode::Unshared const &shadowNode) const override {
+  void adopt(ShadowNode& shadowNode) const override {
     ConcreteComponentDescriptor::adopt(shadowNode);
 
-    auto progressViewhadowNode =
-        std::static_pointer_cast<RNCProgressViewShadowNode>(shadowNode);
+    auto& progressViewhadowNode =
+        static_cast<RNCProgressViewShadowNode&>(shadowNode);
 
     // `RNCProgressViewShadowNode` uses `ImageManager` to initiate image loading
     // and communicate the loading state and results to mounting layer.
-    progressViewhadowNode->setImageManager(imageManager_);
+    progressViewhadowNode.setImageManager(imageManager_);
   }
 
  private:
diff --git a/node_modules/@react-native-community/progress-view/common/cpp/react/renderer/components/progressview/RNCProgressViewShadowNode.h b/node_modules/@react-native-community/progress-view/common/cpp/react/renderer/components/progressview/RNCProgressViewShadowNode.h
index 2bf9858..2945fd9 100644
--- a/node_modules/@react-native-community/progress-view/common/cpp/react/renderer/components/progressview/RNCProgressViewShadowNode.h
+++ b/node_modules/@react-native-community/progress-view/common/cpp/react/renderer/components/progressview/RNCProgressViewShadowNode.h
@@ -29,15 +29,15 @@ public:
   void setImageManager(const SharedImageManager &imageManager);
 
   static RNCProgressViewState initialStateData(
-      ShadowNodeFragment const &fragment,
-      ShadowNodeFamilyFragment const &familyFragment,
-      ComponentDescriptor const &componentDescriptor) {
+      const Props::Shared& props,
+      const ShadowNodeFamily::Shared& family,
+      const ComponentDescriptor& componentDescriptor) {
     auto imageSource = ImageSource{ImageSource::Type::Invalid};
     return {
         imageSource,
-        {imageSource, nullptr},
+        {imageSource, nullptr, SharedFunction<>()},
         imageSource,
-        {imageSource, nullptr},
+        {imageSource, nullptr, SharedFunction<>()},
     };
   }
 
diff --git a/node_modules/@react-native-community/progress-view/react-native-progress-view.podspec b/node_modules/@react-native-community/progress-view/react-native-progress-view.podspec
index 3ddce84..763d9bc 100644
--- a/node_modules/@react-native-community/progress-view/react-native-progress-view.podspec
+++ b/node_modules/@react-native-community/progress-view/react-native-progress-view.podspec
@@ -20,8 +20,6 @@ Pod::Spec.new do |s|
   s.osx.source_files = 'macos/**/*.{h,m}'
 
   if fabric_enabled
-    folly_compiler_flags = '-DFOLLY_NO_CONFIG -DFOLLY_MOBILE=1 -DFOLLY_USE_LIBCPP=1 -Wno-comma -Wno-shorten-64-to-32'
-
     s.subspec 'common' do |ss|
       ss.source_files         = 'common/cpp/**/*.{cpp,h}'
       ss.header_dir           = 'react/renderer/components/progressview'
@@ -29,23 +27,10 @@ Pod::Spec.new do |s|
     end
 
     s.subspec 'fabric' do |ss|
-      s.compiler_flags = folly_compiler_flags + ' -DRCT_NEW_ARCH_ENABLED=1'
-
-      s.pod_target_xcconfig = {
-        'HEADER_SEARCH_PATHS' => '\'$(PODS_ROOT)/boost\'',
-        'OTHER_CPLUSPLUSFLAGS' => '-DFOLLY_NO_CONFIG -DFOLLY_MOBILE=1 -DFOLLY_USE_LIBCPP=1',
-        'CLANG_CXX_LANGUAGE_STANDARD' => 'c++17'
-      }
-
-      s.dependency 'RCT-Folly'
-      s.dependency 'RCTRequired'
-      s.dependency 'RCTTypeSafety'
-      s.dependency 'ReactCommon/turbomodule/core'
-      ss.dependency 'React-Codegen'
-      ss.dependency 'React-RCTFabric'
       ss.dependency 'react-native-progress-view/common'
       ss.source_files         = 'ios/Fabric/**/*.{h,m,mm}'
       ss.pod_target_xcconfig  = { 'HEADER_SEARCH_PATHS' => '\'$(PODS_TARGET_SRCROOT)/common/cpp\'' }
+      install_modules_dependencies(s)
     end
   else
     s.dependency 'React-Core'
