diff --git a/node_modules/react-native-reorderable-list/lib/commonjs/components/ReorderableListCore.js b/node_modules/react-native-reorderable-list/lib/commonjs/components/ReorderableListCore.js
index 2e5bea3..c2fdb27 100644
--- a/node_modules/react-native-reorderable-list/lib/commonjs/components/ReorderableListCore.js
+++ b/node_modules/react-native-reorderable-list/lib/commonjs/components/ReorderableListCore.js
@@ -435,8 +435,10 @@ const ReorderableListCore = ({
         dragXY.value = (0, _reactNativeReanimated.withTiming)(newPositionXY, {
           duration: animationDurationProp.value,
           easing: _reactNativeReanimated.Easing.out(_reactNativeReanimated.Easing.ease)
-        }, () => {
-          (0, _reactNativeReanimated.runOnJS)(reorder)(draggedIndex.value, currentIndex.value);
+        }, finished => {
+          if (finished) {
+            (0, _reactNativeReanimated.runOnJS)(reorder)(draggedIndex.value, currentIndex.value);
+          }
         });
       } else {
         // User might drag and release the item without moving it so,
diff --git a/node_modules/react-native-reorderable-list/lib/module/components/ReorderableListCore.js b/node_modules/react-native-reorderable-list/lib/module/components/ReorderableListCore.js
index 3b394e7..30a31cc 100644
--- a/node_modules/react-native-reorderable-list/lib/module/components/ReorderableListCore.js
+++ b/node_modules/react-native-reorderable-list/lib/module/components/ReorderableListCore.js
@@ -427,8 +427,10 @@ const ReorderableListCore = ({
         dragXY.value = withTiming(newPositionXY, {
           duration: animationDurationProp.value,
           easing: Easing.out(Easing.ease)
-        }, () => {
-          runOnJS(reorder)(draggedIndex.value, currentIndex.value);
+        }, finished => {
+          if (finished) {
+            runOnJS(reorder)(draggedIndex.value, currentIndex.value);
+          }
         });
       } else {
         // User might drag and release the item without moving it so,
diff --git a/node_modules/react-native-reorderable-list/src/components/ReorderableListCore.tsx b/node_modules/react-native-reorderable-list/src/components/ReorderableListCore.tsx
index 908155c..31d5e81 100644
--- a/node_modules/react-native-reorderable-list/src/components/ReorderableListCore.tsx
+++ b/node_modules/react-native-reorderable-list/src/components/ReorderableListCore.tsx
@@ -694,8 +694,10 @@ const ReorderableListCore = <T,>(
               duration: animationDurationProp.value,
               easing: Easing.out(Easing.ease),
             },
-            () => {
-              runOnJS(reorder)(draggedIndex.value, currentIndex.value);
+            finished => {
+              if (finished) {
+                runOnJS(reorder)(draggedIndex.value, currentIndex.value);
+              }
             },
           );
         } else {
